### ISSUE TITLE
[Dashboard] Task table doesn't show task if tasks > 10,000 

### ISSUE THREAD

**Issue by galenhwang** [ray-project/ray#38592]
### What happened + What you expected to happen

https://github.com/ray-project/ray/blob/d171d9ec007bba9b4e583958ed66648f417ae81c/dashboard/client/src/service/task.ts#L6

### Versions / Dependencies

2.6.3

### Reproduction script

n/a

### Issue Severity

None

**Comment by galenhwang** [comment]
if not trivial, it'd be preferred if the task table and its API are paginated:

https://github.com/rkooo567/ray/blob/ae85e8266ee763532b2e1580385aa79aeda6b3a6/dashboard/state_aggregator.py#L379-L386

**Comment by chaowanggg** [comment]
I agree, since we don't have a pagination support in the backend. We could also use a frontend pagination to display all data from backend

cc @alanwguo

**Comment by alanwguo** [comment]
we do not support more than 10,000 tasks. Old tasks get GC'd by ray to conserve memory.

**Comment by alanwguo** [comment]
We plan to introduce proprietary value where we anyscale ray clusters can support more than 10,000 tasks.

To help alleviate some of these issues, in OSS, we are updating our tasks GC logic to keep around tasks with errors for longer and we are updating the UI to show errored tasks above non-errored tasks
